### PR TITLE
[TypeScript][Fetch] Make configuration properties optional

### PR DESCRIPTION
**Pull request by Aleksion**
Right now all properties are required despite the enabled authentication schemes

### PR checklist

- [x] Read the [contribution guildelines](https://github.com/swagger-api/swagger-codegen/blob/master/CONTRIBUTING.md).
- [x] Ran the shell/batch script under `./bin/` to update Petstore sample so that CIs can verify the change. (For instance, only need to run `./bin/{LANG}-petstore.sh` and `./bin/security/{LANG}-petstore.sh` if updating the {LANG} (e.g. php, ruby, python, etc) code generator or {LANG} client's mustache templates)
- [x] Filed the PR against the correct branch: master for non-breaking changes and `2.3.0` branch for breaking (non-backward compatible) changes.

### Description of the PR
Made configuration properties optional in typescript-fetch
(details of the change, additional tests that have been done, reference to the issue for tracking, etc)

### CODE REVIEW

**Comment by wing328** [comment]
@Aleksion thanks for the PR. Please run the following to update the Petstore sample for TS Fetch API client:
```
./bin/typescript-fetch-petstore-all.sh
```
so that CIs can verify the change.

cc @Vrolijkx @leonyu @alechill

File: modules/swagger-codegen/src/main/resources/TypeScript-Fetch/configuration.mustache [hunk 1]
```diff
@@ -1,12 +1,12 @@
...
         {{keyParamName}}: string;
 {{/isApiKey}}
 {{/authMethods}}
     };
-    username: string;
-    password: string;
-    accessToken: string;
-}
\ No newline at end of file
+    usernam?: string;
```
Comment (bherila) [inline modules/swagger-codegen/src/main/resources/TypeScript-Fetch/configuration.mustache]: `"usernam?` is spelled wrong. It should be `username?`

Comment (Aleksion) [inline modules/swagger-codegen/src/main/resources/TypeScript-Fetch/configuration.mustache]: Shit. You're right. Really sorry about that!

**Comment by Aleksion** [comment]
@wing328 - Sorry about that.
It's now been run, and I fixed the typo (impressive I managed to make a mistake with 4 lines of changes >.< Long live Pull Requests)

**Comment by ddunkin** [comment]
Optional class properties are new in TypeScript 2.0, but this package builds with TypeScript 1.8.

https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-0.html#optional-class-properties

**Comment by wing328** [comment]
The latest master has the enhancement in this PR:
```
// tslint:disable
{{>licenseInfo}}

export interface ConfigurationParameters {
    apiKey?: string | ((name: string) => string);
    username?: string;
    password?: string;
    accessToken?: string | ((name: string, scopes?: string[]) => string);
    basePath?: string;
}
```

**Pull request closed without merging**